share/availability/discovery/discovery.go

```diff
@@ -256,6 +264,7 @@ func (d *Discovery) ensurePeers(ctx context.Context) {
 			continue
 		}
 
+		//TODO: drain the channel, because it could be filled already or use timer with proper drain
```

Comment:
Ticked does not have to be drained, as per docs. It accounts for slow reader

Reply:
Yeah, but check the ticker code. If channel was filled it won't be drained on Reset. Doc is not what you expect, because Reset operation is not meant to make a fresh new ticker that will yield first time after delay. It just updates the tick time, while persisting current ticker state. For our usecase we need to drain (could be my next commit)

Reply:
Example code for the ticker that is not properly reset:
```
ticker := time.NewTicker(2 * time.Second)
for {
time.Sleep(2* time.Second )
ticker.Reset(2 * time.Second)

select {
case <-ticker.C:
	fmt.Println("ticker fired too early")
case <-time.After(time.Second):
	fmt.Println("ticker didn't fire too early")
}
}
```